pyccel/ast/operators.py

```diff
@@ -520,10 +520,12 @@ def _calculate_shape(arg1, arg2):
             other = [a for a in args if isinstance(a.dtype, FixedSizeNumericType)]
             assert len(other) == 0
             shape = None
-        else:
+        elif any(isinstance(a.class_type, NumpyNDArrayType) for a in (arg1, arg2)):
             s = broadcast(arg1.shape, arg2.shape)
 
             shape = s
```

Comment:
Unrelated to this PR, but... What is the point of creating the temporary variable `s` here?

Reply:
No idea. Removed: [570b141](https://github.com/pyccel/pyccel/pull/2300/commits/570b141bb6e8105c0eecf62f68858470509752ba)